crates/net/network/src/config.rs

```diff
@@ -33,6 +34,8 @@ pub fn rng_secret_key() -> SecretKey {
 /// All network related initialization settings.
 #[derive(Debug)]
 pub struct NetworkConfig<C, N: NetworkPrimitives = EthNetworkPrimitives> {
+    /// The Ethereum protocol handler
+    pub eth_protocol_handler: Box<dyn DynEthProtocolHandler<N>>,
```

Comment:
for now we can keep the dyn hack, but I think we could even make this an associated type of NetworkPrimitives or a new trait that is also networkprims